Generic webhook request payload v2 docs (#57)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/notification-configurations.mdx

## Notification Payload

> The notification is an HTTP POST request with a detailed payload whose content depends on the type of notification configured.

Reviewer: ```suggestion
The notification is an HTTP POST request with a detailed payload. The content depends on the type of notification.
```

> For Slack and Microsoft Teams notifications, the payload conforms to the respective webhook API, and results in a notification message with informational attachments. Refer to [Slack Notification Payloads](/cloud-docs/workspaces/settings/notifications#slack) and [Microsoft Teams Notification Payloads](/cloud-docs/workspaces/settings/notifications#microsoft-teams)  for examples. For generic notifications, the payload varies based on the type of notification:

Reviewer: ```suggestion
For Slack and Microsoft Teams notifications, the payload conforms to the respective webhook API and results in a notification message with informational attachments. Refer to [Slack Notification Payloads](/cloud-docs/workspaces/settings/notifications#slack) and [Microsoft Teams Notification Payloads](/cloud-docs/workspaces/settings/notifications#microsoft-teams)  for examples. For generic notifications, the payload varies based on whether the notification contains information about run events or workspace events.
```

> ### Run Notification Payload

Reviewer: I'm just thinking it might be nice to have some preamble here to help reinforce the idea that there are two types of events and that this payload stuff is ONLY for generic notifications... idk thoughts?

```suggestion
### Run Notification Payload

Run events include detailed information about a specific run, including the time it began and the associated workspace and organization. Generic notifications for run events contain the following information:
```

Author: Yeah that makes sense to me.

}
```

> ### Workspace Notification Payload

Reviewer: ```suggestion
### Workspace Notification Payload

Workspace events include detailed information about workspace-level validation events like [drift detection](/cloud-docs/workspaces/settings/drift-detection) if it is enabled for the workspace. Much of the information provides details about the associated [assessment result](/cloud-docs/api-docs/assessment-results), which Terraform Cloud uses to track instances of continuous validation. Generic notifications for workspace events contain the following information:
```

| Name                                         | Type   | Description                                                               |
| -------------------------------------------- | ------ | ------------------------------------------------------------------------- |

> | `payload_version`                            | number | Currently always "2".                                                     |

Reviewer: We should avoid time-bound language in the docs if possible :-) 

```suggestion
| `payload_version`                            | number | Always "2".                                                     |
```

Author: This is copied from the run payload above.  I will change it there, too.

> | `notification_configuration_id`              | string | ID of the configuration which defined this notification.                  |

Reviewer: should it be "that triggered this notification" or that "created this notification". Defined feels weird to me as the user defined it. Thoughts?

```suggestion
| `notification_configuration_id`              | string | ID of the configuration that defined this notification.                  |
```

Author: (More language borrowed from the run payload -- will update that with whatever we decide).  I see what you mean, though neither "created" nor "triggered" feels write to me.  The notification configuration doesn't trigger the notification, it specifies the conditions that will trigger it.  "corresponding to"? "from which this notification results" lol? "caused"?  They all feel a little off to me; I don't have a strong opinion.

> | `notification_configuration_url`             | string | URL of the Notification Configuration.                                    |

Reviewer: Does this take you to it in TFC? Where does the URL go?

Reviewer: How about:

"URL to get the notification configuration from the Terraform Cloud API."

> | `trigger_scope`                              | string | "assessment" for workspace notifications.                                 |

Reviewer: I don't know what we mean by "assessment" here - help :-)

Author: This field is intended to support more notification types in v2 in the future.  Drift detection notifications will all have the value `"assessment"`.  If we migrated run notifications to v2, they would have the value `"run"`.  Hypothetical future notification types would have a new value.  But for now, only drift detection notifications use v2, so this field (**for now**) will only even have the value `"assessment"`.  I very much do not like what I came up with here, so please give me better words :)

| `trigger`                                    | string | Value of the trigger which caused the notification to be sent.            |
| `message`                                    | string | Human-readable reason for the notification.                               |
| `details`                                    | object | Object containing details specific to the notification.                   |
| `details.new_assessment_result`              | object | The most recent Assessment Result which triggered the notification.       |

> | `details.new_assessment_result.id`           | string | ID of the Assessment Result.                                              |

Reviewer: Let's not capitalize assessment result. Do we capitalize things like this in other places? In the regular docs, it's best practice that we do not capitalize feature names, etc. But lmk if I'm incorrect here about how we typically do API

Author: Nope, that was unintentional.

| `details.new_assessment_result.url`          | string | URL of the Assessment Result.                                             |
| `details.new_assessment_result.succeeded`    | bool   | Whether or not Assessment succeeded.                                      |
| `details.new_assessment_result.drifted`      | bool   | Whether or not Assessment detected drift.                                 |
| `details.new_assessment_result.created_at`   | string | Timestamp of the Assessment Result                                        |
| `details.prior_assessment_result`            | object | The Assessment Result prior to the one which triggered the notification.  |
| `details.prior_assessment_result.id`         | string | ID of the Assessment Result.                                              |
| `details.prior_assessment_result.url`        | string | URL of the Assessment Result.                                             |
| `details.prior_assessment_result.succeeded`  | bool   | Whether or not Assessment succeeded.                                      |

> | `details.prior_assessment_result.drifted`    | bool   | Whether or not Assessment detected drift.                                 |

Reviewer: This is "prior" --> how is this different from the new one that we have above? I think we should specify that in the description!

| `details.prior_assessment_result.created_at` | string | Timestamp of the Assessment Result.                                       |

> | `details.workspace_id`                       | string | ID of the workspace.                                                      |

Reviewer: ```suggestion
| `details.workspace_id`                       | string | ID of the workspace that generated the notification.                                                      |
```
